src/repository/user.remote.rest.repository.ts

```diff
@@ -0,0 +1,25 @@
+import databaseInterface from './database.interface';
+import UserRemote from '../model/entity/user.remote.type';
+
+export const getUserByName = (database: databaseInterface) => async (username: string): Promise<UserRemote> => {
+  const githubUser = await database.handler.request(
+    'GET /users/{username}',
+    { username },
+  );
+
+  const languages = await database.handler.request(
+    'GET /users/{username}/repos',
+    { username },
+  )
```

Comment:
Can you please clarify why are you using this chain of `.then`?

Reply:
This part of the code fetches data from the user's repositories and filters the data into an array of unique programming languages that the user has used in his projects.

I probably abused the ```then()``` statements but it is also probably cleaner than the imperative approach

Since other members are having trouble understanding this portion of the solution, this would be a good candidate to be in a separate method such as ```getUserLanguages()```